azure/dns-records/variables.tf

```
@@ -0,0 +1,24 @@
+variable "dns_zone" {
+  description = "Name of the DNS zone"
+  type        = string
+}
+
+variable "dns_rg" {
+  description = "Resource Group of the DNS Zone"
+  type        = string
+}
```

It might be a good idea to rename these variables to improve clarity in terms of the purpose. I understood their purpose but I had to read the code to understand exactly.
```suggestion
variable "dns_zone_name" {
  description = "Name of the DNS zone"
  type        = string
}

variable "dns_zone_rg" {
  description = "Resource Group of the DNS Zone"
  type        = string
}
```

---

Changed.